### PR TITLE
remove deprecated `readonly` from `IDeltaManager` and `DeltaManager`

### LINKED ISSUE #8438: Remove remaining `readOnly` implementations after two version releases (0.55)
## Work Item
The deprecated `readOnly` property should be removed from the following files after two version releases (0.55)
1. `common\lib\container-definitions\src\deltas.ts`
2. `packages\loader\container-loader\src\deltaManager.ts`

**Approach**
Remove the property and replace any uses of the property with `readOnlyInfo.readonly`.

#### Steps
- [x] Replace all usages of `readonly` with `readOnlyInfo.readonly` in partner repos.
- [ ] https://github.com/microsoft/FluidFramework/pull/8786

**Additional context**
Issue: https://github.com/microsoft/FluidFramework/issues/8190
PR: https://github.com/microsoft/FluidFramework/pull/8385

### PR DESCRIPTION
**Pull request by scottn12**
This PR removes the `readonly` property from `IDeltaManager` and `DeltaManager` from the `container-definitions` and `container-loader` packages respectively. To replace its functionality use `readOnlyInfo.readonly`. All remaining uses of this property were already replaced in https://github.com/microsoft/FluidFramework/pull/8385. 

Closes #8438

### CODE REVIEW

**Comment by vladsud** [comment]
I do not see a change in DeltaManagerProxy

**Comment by scottn12** [comment]
@vladsud, thank you for pointing that out. I think I missed it because it is not actually marked for deprecation currently. There are no remaining usages of `DeltaManagerProxy.readonly()`, so hopefully we can try to remove it as well in this PR.

**Comment by ssimic2** [comment]
> @vladsud, thank you for pointing that out. I think I missed it because it is not actually marked for deprecation currently. There are no remaining usages of `DeltaManagerProxy.readonly()`, so hopefully we can try to remove it as well in this PR.

You should be able to remove it now. API callers can easily access the same flag via `readOnlyInfo`. Of course, call out these breaking changes.

**Comment by ssimic2** [comment]
It looks like there are still some references around mocks (like MockDeltaManager). When you remove those, see if any test cases depended on it.

**Review by ssimic2 — CHANGES_REQUESTED** [review]
It looks like there are still some references around mocks (like MockDeltaManager). When you remove those, see if any test cases depended on it.

File: common/lib/container-definitions/src/deltas.ts [hunk 1]
```diff
@@ -148,21 +148,6 @@ export interface IDeltaManager<T, U> extends IEventProvider<IDeltaManagerEvents>
...
-     *
-     * We do not differentiate here between no write access to storage vs. host disallowing changes to container -
-     * in all cases container runtime and data stores should respect readonly state and not allow local changes.
-     *
-     * It is undefined if we have not yet established websocket connection
-     * and do not know if user has write access to a file.
-     * @deprecated - use readOnlyInfo
-     */
-    readonly readonly?: boolean;
-
```
Comment (heliocliu) [inline common/lib/container-definitions/src/deltas.ts:165]: to confirm, this was the one we talked about before and you verified has no usages in other internal repos, right?

Comment (scottn12) [reply common/lib/container-definitions/src/deltas.ts:165]: I just double checked, and we do actually have some references in partner repos that need to be taken care of first. Not sure why I thought this was already taken care of. We can revisit this PR after those changes are made.

File: BREAKING.md [hunk 2]
```diff
@@ -13,15 +13,18 @@ There are a few steps you can take to write a good change note and avoid needing
...
 
 ### `MessageType.Save` and code that handled it was removed
 The `Save` operation type was deprecated and has now been removed. This removes `MessageType.Save` from `protocol-definitions`, `save;${string}: ${string}` from `SummarizeReason` in the `container-runtime` package, and `MessageFactory.createSave()` from and `server-test-utils`.
 
 ### Removed `IOdspResolvedUrl.sharingLinkToRedeem`
 The `sharingLinkToRedeem` property is removed from the `IOdspResolvedUrl` interface. The property can be accesed from `IOdspResolvedUrl.shareLinkInfo` instead.
 
+### `readonly` removed from `IDeltaManager`, `DeltaManager`, and `DeltaManagerProxy`
+The `readonly` property was deprecated and has now been removed from `IDeltaManager` from `container-definitions`. Additionally, `readonly` has been removed from the implementations in `DeltaManager` and `DeltaManagerProxy` from `container-loader`. To replace its functionality, use `readOnlyInfo.readonly` instead.
+
```
Comment (heliocliu) [inline BREAKING.md:26]:
```suggestion
### readonly removed from IDeltaManager, DeltaManager, and DeltaManagerProxy
The `readonly` property was deprecated and has now been removed from `IDeltaManager` from `container-definitions`. Additionally, `readonly` has been removed from the implementations in `DeltaManager` and `DeltaManagerProxy` from `container-loader`. To replace its functionality, use `readOnlyInfo.readonly` instead.

```

we had some feedback that the number of inline code blocks was making these hard to read.  let's start with not using code blocks in the headers and outside of core "X deprecated from Y, use Z instead" statements in the body and see how that goes?  (so just a header change in this case)

**Review by heliocliu — APPROVED** [review]

**Review by ssimic2 — APPROVED** [review]

**Pull request merged**